pkg/api/types.go

```diff
@@ -1569,72 +1569,26 @@ func (k *KubernetesConfig) GetAddonScript(addonName string) string {
 	return kubeAddon.Data
 }
 
-// isAddonEnabled checks whether a k8s addon with name "addonName" is enabled or not based on the Enabled field of KubernetesAddon.
-// If the value of Enabled in nil, the "defaultValue" is returned.
-func (k *KubernetesConfig) isAddonEnabled(addonName string, defaultValue bool) bool {
+// IsAddonEnabled checks whether a k8s addon with name "addonName" is enabled or not based on the Enabled field of KubernetesAddon.
+// If the value of Enabled is nil, the "defaultValue" is returned.
+func (k *KubernetesConfig) IsAddonEnabled(addonName string) bool {
 	kubeAddon := k.GetAddonByName(addonName)
-	return kubeAddon.IsEnabled(defaultValue)
-}
-
-// IsMetricsServerEnabled checks if the metrics server addon is enabled
-func (o *OrchestratorProfile) IsMetricsServerEnabled() bool {
```

Comment:
There are a bunch of superflous addons-specific methods here, some of which carry their own "default requirements" separate from the main default enforcements flow. Only those methods that are serving functional purposes (e.g., for template-layer functionality) are retained.